x-pack/plugins/uptime/public/hooks/use_url_params.ts

```diff
@@ -72,28 +66,32 @@ export const useUrlParams: UptimeUrlParamsHook = () => {
         ...updatedParams,
       };
 
-      history.push({
-        pathname,
-        search: stringify(
-          // drop any parameters that have no value
-          Object.keys(mergedParams).reduce((params, key) => {
-            const value = mergedParams[key];
-            if (value === undefined || value === '') {
-              return params;
-            }
-            return {
-              ...params,
-              [key]: value,
-            };
-          }, {}),
-          { sort: false }
-        ),
-      });
+      const updatedSearch = stringify(
+        // drop any parameters that have no value
+        Object.keys(mergedParams).reduce((params, key) => {
+          const value = mergedParams[key];
+          if (value === undefined || value === '') {
+            return params;
+          }
+          return {
+            ...params,
+            [key]: value,
+          };
+        }, {})
+      );
+
+      // only update the URL if the search has actually changed
+      if (search !== updatedSearch) {
```

Comment:
How would you feel about updating this to a useEffect which automatically fires when the dependencies change? Uncertain if that is truly necessary this if statement strikes me as something useEffect solves for well.

Reply:
++ actually I was thinking the same thing when I was writing the code. I think since this code is already nested in a hook callback I balked from it but I can revisit.